src/core/net/dnssd_server.cpp

```diff
@@ -547,10 +566,20 @@ Error Server::FindNameComponents(const char *aName, const char *aDomain, NameCom
 
     aInfo.mServiceOffset = labelBegin;
 
-    // Treat everything before <Service> as <Instance> label
+    // Check for service subtype
     error = FindPreviousLabel(aName, labelBegin, labelEnd);
     VerifyOrExit(error == kErrorNone, error = (error == kErrorNotFound ? kErrorNone : error));
 
+    if ((labelEnd == labelBegin + kSubTypeLabelLength) &&
+        (memcmp(&aName[labelBegin], kDnssdSubTypeLabel, kSubTypeLabelLength) == 0))
+    {
+        SuccessOrExit(error = FindPreviousLabel(aName, labelBegin, labelEnd));
+        VerifyOrExit(labelBegin == 0, error = kErrorInvalidArgs);
+        aInfo.mSubTypeOffset = labelBegin;
```

Comment:
Is this new field used anywhere?

Reply:
You are correct. It is not used currently, but since we already detect and parse the subtype label I added it (it may be useful if we later move the code that breaks the name into components and define it a common DNS name utility helper).